services/scabbard/src/cli/main.rs

```diff
@@ -0,0 +1,658 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#[macro_use]
+extern crate log;
+
+mod error;
+mod key;
+mod transaction;
+
+use std::fs::File;
+use std::io::{BufReader, Read};
+
+use clap::{App, AppSettings, Arg, SubCommand};
+use flexi_logger::{DeferredNow, LogSpecBuilder, Logger};
+use log::Record;
+use sabre_sdk::protocol::payload::{
+    Action, CreateContractActionBuilder, CreateContractRegistryActionBuilder,
+    CreateNamespaceRegistryActionBuilder, CreateNamespaceRegistryPermissionActionBuilder,
+    CreateSmartPermissionActionBuilder, DeleteContractRegistryActionBuilder,
+    DeleteNamespaceRegistryActionBuilder, DeleteNamespaceRegistryPermissionActionBuilder,
+    DeleteSmartPermissionActionBuilder, ExecuteContractActionBuilder, SabrePayloadBuilder,
+    UpdateContractRegistryOwnersActionBuilder, UpdateNamespaceRegistryOwnersActionBuilder,
+    UpdateSmartPermissionActionBuilder,
+};
+use sawtooth_sdk::signing::secp256k1::Secp256k1Context;
+use splinter::{
+    service::scabbard::client::{SabreSmartContractDefinition, ScabbardClient},
+    signing::sawtooth::SawtoothSecp256k1RefSigner,
+};
+
+use error::CliError;
+
+fn main() {
+    if let Err(e) = run() {
+        error!("ERROR: {}", e);
+        std::process::exit(1);
+    }
+}
+
+fn run() -> Result<(), CliError> {
+    let app = App::new("scabbard")
+        .version(env!("CARGO_PKG_VERSION"))
+        .author("Cargill")
+        .about("Command line for scabbard")
+        .setting(AppSettings::SubcommandRequiredElseHelp)
+        .arg(
```

Comment:
Can you make these top level argument flags global so they can be placed anywhere in the line

Reply:
Updated all of the optional top-level arguments to be global (required arguments cannot be made global).